segment_images.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import csv
+import os
+from pathlib import Path
+
+import cv2  # type: ignore
+import numpy as np
+import torch
+from segment_anything import SamAutomaticMaskGenerator, sam_model_registry
+
+parser = argparse.ArgumentParser(description="A script that segments multiple images")
+parser.add_argument(
+    "--sam_model_type",
+    help="Segment-Anything model type.",
+    choices=["vit_h", "vit_l", "vit_b"],
+    default="vit_h",
+)
+parser.add_argument(
+    "--checkpoint",
+    help="Path to SAM checkpoint.",
+    default="D:/sam_checkpoints/sam_vit_h_4b8939.pth",
+)
+parser.add_argument(
+    "--points_per_side",
+    type=int,
+    help="The number of points to be sampled along one side of the image. "
+    "The total number of points is points_per_side**2.",
+    default=32,
+)
+parser.add_argument(
+    "--data_path",
+    help="Path to the tensorflow_datasets folder",
+    default="D:/GithubProjects/tensorflow_datasets",
+)
+parser.add_argument("--size", type=int, help="Number of images to segment", default=3)
+parser.add_argument(
+    "--dataset",
+    help="Which dataset to use.",
+    default="cifar10",
+    choices=["cifar10", "mnist", "oxford_flowers102", "imagenet2012"],
+)
+parser.add_argument(
+    "--split",
+    help="Which dataset split to use.",
+    default="test",
+)
+
+parser.add_argument(
+    "--output",
+    help="Path to where the output will be stored",
+    default="./outputs",
+)
+
+
+def register_sam(
+    model_type="vit_h",
+    checkpoint_path="D:/sam_checkpoints/sam_vit_h_4b8939.pth",
+    device="cuda",
+    points_per_side=32,
+):
+    DEVICE = torch.device(device if torch.cuda.is_available() else "cpu")
+    sam = sam_model_registry[model_type](checkpoint=checkpoint_path)
+    sam.to(device=DEVICE)
+    mask_generator = SamAutomaticMaskGenerator(sam, points_per_side=points_per_side)
+    return mask_generator, sam, DEVICE
+
+
+def compute_masks(image, name, generator: SamAutomaticMaskGenerator):
+    """Segments an image into masks using SAM-model"""
+    print(f"Segmenting image {name}...")
```

Comment:
Would move this to the calling side. Unnecessary to specify name just for printing